access/slackbot/main.go

```diff
@@ -225,6 +226,16 @@ func (a *App) Run(ctx context.Context) error {
 
 	ctx, cancel := context.WithCancel(ctx)
 	a.cancel = cancel
+	
+	clientCert, err := x509.ParseCertificate(a.tlsConf.Certificates[0].Certificate[0])
```

Comment:
I don't feel it's the right place to do this check. Here, we must not assume that certificate is located in `Certificates[0]`. Having arbitrary `tls.Config`, we cannot rely on the certificate chain length.

Instead of this, I'd rather check the certificate when it's being loaded from the file.

Reply:
It's also good to save the result of `x509.ParseCertificate` to the `Leaf` field of the certificate. Having this, `tls` implementation won't parse this data twice.